include/unifex/v2/async_scope.hpp

```diff
@@ -264,18 +264,18 @@ struct _nest_receiver<Sender, Receiver>::type final {
   void set_value(T... values) noexcept {
     complete([&](auto&& receiver) noexcept {
       UNIFEX_TRY {
-        unifex::set_value(std::move(receiver), std::move(values)...);
+        unifex::set_value(std::forward<decltype(receiver)>(receiver), std::move(values)...);
       }
       UNIFEX_CATCH(...) {
-        unifex::set_error(std::move(receiver), std::current_exception());
+        unifex::set_error(std::forward<decltype(receiver)>(receiver), std::current_exception());
       }
     });
   }
 
   template <typename E>
   void set_error(E e) noexcept {
     complete([&](auto&& receiver) noexcept {
-      unifex::set_error(std::move(receiver), std::move(e));
+      unifex::set_error(std::forward<decltype(receiver)>(receiver), std::move(e));
```

Comment:
This is semi debatable. In the letter of the law (the law being [CppCoreGuidelines F.19](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-forward) checked by [bugprone-move-forwarding-reference](https://releases.llvm.org/11.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-move-forwarding-reference.html), this should use a `forward`. I think receivers usually or always are passed by rvalue reference within the `set_error` CPO, but if someone somehow ends up trying to pass an lvalue ref to a receiver that does or doesn't accept by lvalue ref on the receiver, this code is not taking an opinion on way or the other - the bug would be elsewhere.